### PR TITLE
[New Rule] Email Reported by User as Malware or Phish

### LINKED ISSUE #1698: [New Rule] Email Reported by User as Malware or Phish
## Description

Detects the occurrence of emails reported as Phishing or Malware by Users. Security Awareness training is essential to
stay ahead of scammers and threat actors, as security products can be bypassed, and the user can still receive a
malicious message. Educating users to report suspicious messages can help identify gaps in security controls and prevent
malware infections and Business Email Compromise attacks.

## Example Data

```
{
  "_index": ".ds-logs-o365.audit-default-2022.01.04-000001",
  "_type": "_doc",
  "_id": "d83ad60c-f49b-4756-c800-08d9d5cb70f7",
  "_version": 1,
  "_score": 1,
  "_ignored": [
    "event.original"
  ],
  "_source": {
    "agent": {
      "hostname": "Shipper",
      "name": "Shipper",
      "id": "21d0ccfd-4784-4126-8445-cc4c3437e6e4",
      "ephemeral_id": "684b66ad-a186-417e-b4a5-eb37263289d1",
      "type": "filebeat",
      "version": "7.16.2"
    },
    "elastic_agent": {
      "id": "21d0ccfd-4784-4126-8445-cc4c3437e6e4",
      "version": "7.16.2",
      "snapshot": false
    },
    "rule": {
      "reference": [
        ""
      ],
      "name": "Email reported by user as malware or phish",
      "id": "b26a5770-0c38-434a-9380-3a3c2c27bbb3",
      "category": "ThreatManagement"
    },
    "message": "New alert",
    "tags": [
      "forwarded",
      "o365-audit",
      "preserve_original_event"
    ],
    "cloud": {
      "instance": {
        "name": "Shipper",
        "id": "3dc43ef6-ee8b-4e06-895b-d134cb4513c8"
      },
      "provider": "azure",
      "machine": {
        "type": "Standard_D2s_v3"
      },
      "service": {
        "name": "Virtual Machines"
      },
      "region": "eastus",
      "account": {}
    },
    "o365": {
      "audit": {
        "Status": "Active",
        "ResultStatus": "Succeeded",
        "ObjectId": "6ea6b45c-e611-1104-c800-08d9d5cb70f7",
        "UserKey": "SecurityComplianceAlerts",
        "Data": "{\"f3u\":\"TestUser@TestOrg.onmicrosoft.com\",\"ts\":\"2022-01-12T12:45:00.0000000Z\",\"te\":\"2022-01-12T13:00:00.0000000Z\",\"op\":\"UserSubmission\",\"wl\":\"SecurityComplianceCenter\",\"tid\":\"1c04d22a-c0e8-4bec-840b-f1a7fb855093\",\"tdc\":\"1\",\"reid\":\"e1ec3e60-b431-4ffb-83ed-08d9d5ca42ca\",\"rid\":\"22964e06-3b21-4508-b7d6-bffa1b66a6ff\",\"cid\":\"b26a5770-0c38-434a-9380-3a3c2c27bbb3\",\"ad\":\"This alert is triggered when any email message is reported as malware or phish by users -V1.0.0.3\",\"lon\":\"UserSubmission\",\"an\":\"Email reported by user as malware or phish\",\"sev\":\"Low\"}",
        "Severity": "Low",
        "Source": "Office 365 Security & Compliance",
        "AlertType": "System",
        "AlertId": "6ea6b45c-e611-1104-c800-08d9d5cb70f7",
        "RecordType": "40",
        "Version": "1",
        "UserId": "SecurityComplianceAlerts",
        "CreationTime": "2022-01-12T12:55:34",
        "UserType": "4"
      }
    },
    "input": {
      "type": "o365audit"
    },
    "@timestamp": "2022-01-12T12:55:34.000Z",
    "ecs": {
      "version": "1.12.0"
    },
    "data_stream": {
      "namespace": "default",
      "type": "logs",
      "dataset": "o365.audit"
    },
    "organization": {
      "id": "1c04d22a-c0e8-4bec-840b-f1a7fb855093"
    },
    "host": {
      "id": "1c04d22a-c0e8-4bec-840b-f1a7fb855093"
    },
    "event": {
      "agent_id_status": "verified",
      "ingested": "2022-01-12T13:03:30Z",
      "original": "{\"CreationTime\":\"2022-01-12T12:55:34\",\"Id\":\"d83ad60c-f49b-4756-c800-08d9d5cb70f7\",\"Operation\":\"AlertTriggered\",\"OrganizationId\":\"1c04d22a-c0e8-4bec-840b-f1a7fb855093\",\"RecordType\":40,\"ResultStatus\":\"Succeeded\",\"UserKey\":\"SecurityComplianceAlerts\",\"UserType\":4,\"Version\":1,\"Workload\":\"SecurityComplianceCenter\",\"ObjectId\":\"6ea6b45c-e611-1104-c800-08d9d5cb70f7\",\"UserId\":\"SecurityComplianceAlerts\",\"AlertId\":\"6ea6b45c-e611-1104-c800-08d9d5cb70f7\",\"AlertLinks\":[{\"AlertLinkHref\":\"\"}],\"AlertType\":\"System\",\"Category\":\"ThreatManagement\",\"Comments\":\"New alert\",\"Data\":\"{\\\"f3u\\\":\\\"TestUser@TestOrg.onmicrosoft.com\\\",\\\"ts\\\":\\\"2022-01-12T12:45:00.0000000Z\\\",\\\"te\\\":\\\"2022-01-12T13:00:00.0000000Z\\\",\\\"op\\\":\\\"UserSubmission\\\",\\\"wl\\\":\\\"SecurityComplianceCenter\\\",\\\"tid\\\":\\\"1c04d22a-c0e8-4bec-840b-f1a7fb855093\\\",\\\"tdc\\\":\\\"1\\\",\\\"reid\\\":\\\"e1ec3e60-b431-4ffb-83ed-08d9d5ca42ca\\\",\\\"rid\\\":\\\"22964e06-3b21-4508-b7d6-bffa1b66a6ff\\\",\\\"cid\\\":\\\"b26a5770-0c38-434a-9380-3a3c2c27bbb3\\\",\\\"ad\\\":\\\"This alert is triggered when any email message is reported as malware or phish by users -V1.0.0.3\\\",\\\"lon\\\":\\\"UserSubmission\\\",\\\"an\\\":\\\"Email reported by user as malware or phish\\\",\\\"sev\\\":\\\"Low\\\"}\",\"Name\":\"Email reported by user as malware or phish\",\"PolicyId\":\"b26a5770-0c38-434a-9380-3a3c2c27bbb3\",\"Severity\":\"Low\",\"Source\":\"Office 365 Security & Compliance\",\"Status\":\"Active\"}",
      "code": "SecurityComplianceAlerts",
      "provider": "SecurityComplianceCenter",
      "kind": "alert",
      "action": "AlertTriggered",
      "id": "d83ad60c-f49b-4756-c800-08d9d5cb70f7",
      "type": [
        "info"
      ],
      "category": [
        "web"
      ],
      "dataset": "o365.audit",
      "outcome": "success"
    },
    "user": {
      "id": "SecurityComplianceAlerts"
    }
  },
  "fields": {
    "rule.id": [
      "b26a5770-0c38-434a-9380-3a3c2c27bbb3"
    ],
    "elastic_agent.version": [
      "7.16.2"
    ],
    "event.category": [
      "web"
    ],
    "o365.audit.Data": [
      "{\"f3u\":\"TestUser@TestOrg.onmicrosoft.com\",\"ts\":\"2022-01-12T12:45:00.0000000Z\",\"te\":\"2022-01-12T13:00:00.0000000Z\",\"op\":\"UserSubmission\",\"wl\":\"SecurityComplianceCenter\",\"tid\":\"1c04d22a-c0e8-4bec-840b-f1a7fb855093\",\"tdc\":\"1\",\"reid\":\"e1ec3e60-b431-4ffb-83ed-08d9d5ca42ca\",\"rid\":\"22964e06-3b21-4508-b7d6-bffa1b66a6ff\",\"cid\":\"b26a5770-0c38-434a-9380-3a3c2c27bbb3\",\"ad\":\"This alert is triggered when any email message is reported as malware or phish by users -V1.0.0.3\",\"lon\":\"UserSubmission\",\"an\":\"Email reported by user as malware or phish\",\"sev\":\"Low\"}"
    ],
    "rule.reference": [
      ""
    ],
    "o365.audit.UserId": [
      "SecurityComplianceAlerts"
    ],
    "o365.audit.Status": [
      "Active"
    ],
    "agent.name": [
      "Shipper"
    ],
    "event.agent_id_status": [
      "verified"
    ],
    "event.kind": [
      "alert"
    ],
    "event.outcome": [
      "success"
    ],
    "user.id": [
      "SecurityComplianceAlerts"
    ],
    "cloud.region": [
      "eastus"
    ],
    "rule.name": [
      "Email reported by user as malware or phish"
    ],
    "input.type": [
      "o365audit"
    ],
    "agent.hostname": [
      "Shipper"
    ],
    "o365.audit.ObjectId": [
      "6ea6b45c-e611-1104-c800-08d9d5cb70f7"
    ],
    "data_stream.type": [
      "logs"
    ],
    "tags": [
      "forwarded",
      "o365-audit",
      "preserve_original_event"
    ],
    "o365.audit.Source": [
      "Office 365 Security & Compliance"
    ],
    "event.provider": [
      "SecurityComplianceCenter"
    ],
    "cloud.provider": [
      "azure"
    ],
    "cloud.machine.type": [
      "Standard_D2s_v3"
    ],
    "event.code": [
      "SecurityComplianceAlerts"
    ],
    "cloud.service.name": [
      "Virtual Machines"
    ],
    "agent.id": [
      "21d0ccfd-4784-4126-8445-cc4c3437e6e4"
    ],
    "ecs.version": [
      "1.12.0"
    ],
    "o365.audit.Severity": [
      "Low"
    ],
    "o365.audit.RecordType": [
      "40"
    ],
    "organization.id": [
      "1c04d22a-c0e8-4bec-840b-f1a7fb855093"
    ],
    "agent.version": [
      "7.16.2"
    ],
    "o365.audit.CreationTime": [
      "2022-01-12T12:55:34"
    ],
    "o365.audit.UserKey": [
      "SecurityComplianceAlerts"
    ],
    "o365.audit.Version": [
      "1"
    ],
    "cloud.instance.id": [
      "3dc43ef6-ee8b-4e06-895b-d134cb4513c8"
    ],
    "agent.type": [
      "filebeat"
    ],
    "o365.audit.AlertType": [
      "System"
    ],
    "event.module": [
      "o365"
    ],
    "elastic_agent.snapshot": [
      false
    ],
    "host.id": [
      "1c04d22a-c0e8-4bec-840b-f1a7fb855093"
    ],
    "elastic_agent.id": [
      "21d0ccfd-4784-4126-8445-cc4c3437e6e4"
    ],
    "data_stream.namespace": [
      "default"
    ],
    "message": [
      "New alert"
    ],
    "o365.audit.UserType": [
      "4"
    ],
    "o365.audit.ResultStatus": [
      "Succeeded"
    ],
    "event.ingested": [
      "2022-01-12T13:03:30.000Z"
    ],
    "event.action": [
      "AlertTriggered"
    ],
    "@timestamp": [
      "2022-01-12T12:55:34.000Z"
    ],
    "o365.audit.AlertId": [
      "6ea6b45c-e611-1104-c800-08d9d5cb70f7"
    ],
    "event.type": [
      "info"
    ],
    "data_stream.dataset": [
      "o365.audit"
    ],
    "agent.ephemeral_id": [
      "684b66ad-a186-417e-b4a5-eb37263289d1"
    ],
    "event.id": [
      "d83ad60c-f49b-4756-c800-08d9d5cb70f7"
    ],
    "rule.category": [
      "ThreatManagement"
    ],
    "event.dataset": [
      "o365.audit"
    ],
    "cloud.instance.name": [
      "Shipper"
    ]
  },
  "ignored_field_values": {
    "event.original": [
      "{\"CreationTime\":\"2022-01-12T12:55:34\",\"Id\":\"d83ad60c-f49b-4756-c800-08d9d5cb70f7\",\"Operation\":\"AlertTriggered\",\"OrganizationId\":\"1c04d22a-c0e8-4bec-840b-f1a7fb855093\",\"RecordType\":40,\"ResultStatus\":\"Succeeded\",\"UserKey\":\"SecurityComplianceAlerts\",\"UserType\":4,\"Version\":1,\"Workload\":\"SecurityComplianceCenter\",\"ObjectId\":\"6ea6b45c-e611-1104-c800-08d9d5cb70f7\",\"UserId\":\"SecurityComplianceAlerts\",\"AlertId\":\"6ea6b45c-e611-1104-c800-08d9d5cb70f7\",\"AlertLinks\":[{\"AlertLinkHref\":\"\"}],\"AlertType\":\"System\",\"Category\":\"ThreatManagement\",\"Comments\":\"New alert\",\"Data\":\"{\\\"f3u\\\":\\\"TestUser@TestOrg.onmicrosoft.com\\\",\\\"ts\\\":\\\"2022-01-12T12:45:00.0000000Z\\\",\\\"te\\\":\\\"2022-01-12T13:00:00.0000000Z\\\",\\\"op\\\":\\\"UserSubmission\\\",\\\"wl\\\":\\\"SecurityComplianceCenter\\\",\\\"tid\\\":\\\"1c04d22a-c0e8-4bec-840b-f1a7fb855093\\\",\\\"tdc\\\":\\\"1\\\",\\\"reid\\\":\\\"e1ec3e60-b431-4ffb-83ed-08d9d5ca42ca\\\",\\\"rid\\\":\\\"22964e06-3b21-4508-b7d6-bffa1b66a6ff\\\",\\\"cid\\\":\\\"b26a5770-0c38-434a-9380-3a3c2c27bbb3\\\",\\\"ad\\\":\\\"This alert is triggered when any email message is reported as malware or phish by users -V1.0.0.3\\\",\\\"lon\\\":\\\"UserSubmission\\\",\\\"an\\\":\\\"Email reported by user as malware or phish\\\",\\\"sev\\\":\\\"Low\\\"}\",\"Name\":\"Email reported by user as malware or phish\",\"PolicyId\":\"b26a5770-0c38-434a-9380-3a3c2c27bbb3\",\"Severity\":\"Low\",\"Source\":\"Office 365 Security & Compliance\",\"Status\":\"Active\"}"
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Resolves #1698 

## Summary

Detects the occurrence of emails reported as Phishing or Malware by Users. Security Awareness training is essential to
stay ahead of scammers and threat actors, as security products can be bypassed, and the user can still receive a
malicious message. Educating users to report suspicious messages can help identify gaps in security controls and prevent
malware infections and Business Email Compromise attacks.

### CODE REVIEW

**Review by Samirbous — APPROVED** [review]

File: rules/integrations/o365/initial_access_o365_user_reported_phish_malware.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+stay ahead of scammers and threat actors, as security products can be bypassed, and the user can still receive a
+malicious message. Educating users to report suspicious messages can help identify gaps in security controls and prevent
+malware infections and Business Email Compromise attacks.
+"""
+false_positives = ["Legitimate files reported by the users"]
+from = "now-30m"
+index = ["filebeat-*", "logs-o365*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Email Reported by User as Malware or Phish"
```
Comment (brokensound77) [inline rules/integrations/o365/initial_access_o365_user_reported_phish_malware.toml:20]: Should add O365 somewhere in here to be consistent with the other integration rules. (Actually all of the rules for this integration need the names normalized)

File: rules/integrations/o365/initial_access_o365_user_reported_phish_malware.toml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+]
+risk_score = 47
+rule_id = "5930658c-2107-4afc-91af-e0e55b7f7184"
+severity = "medium"
+tags = ["Elastic", "Cloud", "Microsoft 365", "Continuous Monitoring", "SecOps", "Initial Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:o365.audit and event.provider:SecurityComplianceCenter and event.action:AlertTriggered  and rule.name : "Email reported by user as malware or phish"
```
Comment (brokensound77) [inline rules/integrations/o365/initial_access_o365_user_reported_phish_malware.toml:35]:
```suggestion
event.dataset:o365.audit and event.provider:SecurityComplianceCenter and event.action:AlertTriggered and rule.name:"Email reported by user as malware or phish"
```

**Review by brokensound77 — APPROVED** [review]
A few comments, then ✅

File: rules/integrations/o365/initial_access_o365_user_reported_phish_malware.toml (hunk 1 above)
Comment (w0rk3r) [reply rules/integrations/o365/initial_access_o365_user_reported_phish_malware.toml]: Added it to the start

**Pull request merged**